docroot/modules/custom/civictheme_migration/src/Form/CivicThemeMigrationConfigurationForm.php

```diff
@@ -0,0 +1,500 @@
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    if ($this->isGenerateMigrationSubmit($form_state)) {
+      $files = $form_state->getValue('configuration_files');
+      if (empty($files)) {
+        $form_state->setError($form['configuration_files'], $this->t('Configuration files are required to generate a migration'));
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $config = $this->config('civictheme_migration.settings');
+    if ($this->isRetrieveFilesSubmit($form_state)) {
+      $urls = explode("\r\n", $form_state->getValue('endpoint'));
+      try {
+        $files = $this->retrieveFiles($urls);
+        $existing_files = $config->get('configuration_files') ?? [];
+        $config->set('configuration_files', array_merge($existing_files, $files));
+        $this->messenger()->addStatus($this->t('Merlin extracted content JSON files have been retrieved'));
+      }
+      catch (\Exception $exception) {
+        $this->messenger()->addError($exception->getMessage());
+      }
+    }
+
+    if ($this->isSaveConfigurationSubmit($form_state) || $this->isGenerateMigrationSubmit($form_state)) {
+      $config->set('configuration_files', $form_state->getValue('configuration_files'));
+    }
+    $config->set('migration_type', $form_state->getValue('migration_type'));
+    $config->set('remote', [
+      'auth_type' => $form_state->getValue('auth_type'),
+      'auth_username' => $form_state->getValue('auth_username'),
+      'auth_password' => $form_state->getValue('auth_password'),
+      'auth_token' => $form_state->getValue('auth_token'),
+      'endpoint' => $form_state->getValue('endpoint'),
+    ]);
+    $config->save();
+    $this->messenger()->addStatus($this->t('The configuration options have been saved.'));
+
+    if ($this->isGenerateMigrationSubmit($form_state)) {
+      $this->generateMigrationConfigurationSubmit($form, $form_state);
+    }
+
+  }
+
+  /**
+   * Handles submission to generate and import migration configuration.
+   *
+   * @param array $form
+   *   Form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Formstate object.
+   */
+  protected function generateMigrationConfigurationSubmit(array &$form, FormStateInterface $form_state) {
+    $file_ids = $form_state->getValue('configuration_files');
+    $files = $this->fileStorage->loadMultiple($file_ids);
+    $file_steam_wrappers = [];
+    foreach ($files as $file) {
+      $file_steam_wrappers[] = $file->getFileUri();
+    }
+    $migration_config_file_name = $this->extensionPathResolver->getPath('module', 'civictheme_migration') . '/assets/migrate_plus.migration.civictheme_page.yml';
+    $migration_config = file_get_contents($migration_config_file_name);
+    $migration_config = $this->yaml->decode($migration_config);
```

Comment:
Didn't need to do any scrappy `file_get_contents` and `str_replace` - migration files are YML so we can load, decode, update and encode.